### DISCUSSION TITLE
How to create theme in lit dev

### DISCUSSION THREAD

**Discussion by jitendraP-ashutec** [lit/lit#3019] [Q&A]
How to create theme base lit dev component? can I create only styles element which I can export and can use directly those class i n  build in element?

**Comment by ynxtynxt** [comment]
You can export as css template literal tags and adopt them by including them in the `static styles` array within any element .

https://lit.dev/docs/components/styles/#sharing-styles

**Reply by jitendraP-ashutec** [reply]
@ynxtynxt , Thanks for your response, But here my scenario is different. Let me explain you what I want to try to create.

Normally we include styles.css in our project and use those classes into our html pages. same thing I want to create using lit.dev. 
want to create independent styles which I can include in any html file and use those classes.

For Example: I have one html file where I have imported custom-styles.js which export only css classes,  and I want to use those class  build in element. as you can see in below example.

index.html
```
 <script type="module" src="../custom-styles.js"></script>

// corp-button is defined in custom-styles.js and I want to use it in out side of lit element/ any html file.
<button  class="corp-button">  Value </button>

```

Thanks,
Jitendra Prajapati

**Reply by ynxtynxt** [reply]
What custom-styles.js are exporting? They are not some css template literal tags?

**Reply by jitendraP-ashutec** [reply]
@ynxtynxt  custom-styles.js will be like:

```

import { css } from 'lit';
import { LitElement } from 'lit-element';

export class WcTheme extends LitElement {
  static override styles = [
    css`
          .corp-button {
            background-color: #e7e7e7;
            color: black;
            border: none;
            border-radius: 4px;
            padding: 15px 32px;
            font-size: 16px;
            margin: 4px 2px;
            cursor: pointer;
          }
          .corp-button:hover {
            background-color: #aaaaaa;
          }
        `
  ];
}
```

**Reply by ynxtynxt** [reply]
Then export only the css and wrap all the elements with a custom element and import the css into it?

**Reply by jitendraP-ashutec** [reply]
@ynxtynxt  I am not sure, I understand your answer, could you please give example of it ?

**Reply by williamjacobszoon** [reply]
@jitendraP-ashutec You need to use webpack then to archive that. Use the css-loader to get the css mapped into a javascript variable and then use the static style property as the documentation suggested. Following stack overflow answer might give you an idea.

[How to import CSS modules with Typescript, React and Webpack](https://stackoverflow.com/questions/41336858/how-to-import-css-modules-with-typescript-react-and-webpack)

**Reply by ynxtynxt** [reply]
I mean, just wrap all the native elements into a single or more lit-elements and place them in `<body>` so you can have all your theming with css template literal tag. I don't know if it's recommended to use exteral stylesheet. 
[playground](https://lit.dev/playground/#project=W3sibmFtZSI6InNpbXBsZS1ib2R5LmpzIiwiY29udGVudCI6ImltcG9ydCB7aHRtbCwgY3NzLCBMaXRFbGVtZW50fSBmcm9tICdsaXQnO1xuaW1wb3J0IHt1bnNhZmVIVE1MfSBmcm9tICdsaXQvZGlyZWN0aXZlcy91bnNhZmUtaHRtbC5qcyc7XG5pbXBvcnQgeyB1bnRpbCB9IGZyb20gJ2xpdC9kaXJlY3RpdmVzL3VudGlsLmpzJztcbmltcG9ydCB7IGNvcnBTdHlsZXMgfSBmcm9tICcuL2NvcnAtc3R5bGVzLmpzJztcblxuZXhwb3J0IGNsYXNzIFNpbXBsZUJvZHkgZXh0ZW5kcyBMaXRFbGVtZW50IHtcbiAgc3RhdGljIHN0eWxlcyA9IFtjb3JwU3R5bGVzLCBjc3NgXG4gICAgICAgICAgICAgICBcbiAgICAgIDpob3N0IHtcbiAgICAgIGRpc3BsYXk6IGNvbnRlbnRzO1xuICAgICAgfVxuICAgICAgYF07XG5cblxuICByZW5kZXIoKSB7XG4gICAgcmV0dXJuIGh0bWxgXG4gICAgICAgICA8YnV0dG9uICBjbGFzcz1cImNvcnAtYnV0dG9uXCI-ICBWYWx1ZSA8L2J1dHRvbj5cbiAgICAgICAgIDxidXR0b24gIGNsYXNzPVwiY29ycC1idXR0b25cIj4gIFZhbHVlIDwvYnV0dG9uPlxuICAgICAgICAgPGJ1dHRvbiAgY2xhc3M9XCJjb3JwLWJ1dHRvblwiPiAgVmFsdWUgPC9idXR0b24-XG4gICAgICAgIGA7XG4gIH0gICAgXG59XG5jdXN0b21FbGVtZW50cy5kZWZpbmUoJ3NpbXBsZS1ib2R5JywgU2ltcGxlQm9keSk7XG4ifSx7Im5hbWUiOiJpbmRleC5odG1sIiwiY29udGVudCI6IjwhRE9DVFlQRSBodG1sPlxuPGhlYWQ-XG4gIDxzY3JpcHQgdHlwZT1cIm1vZHVsZVwiIHNyYz1cIi4vc2ltcGxlLWJvZHkuanNcIj48L3NjcmlwdD5cbjxzdHlsZT5cbiAgYm9keSB7IGJhY2tncm91bmQtY29sb3I6IGdyZWVuIH1cbjwvc3R5bGU-XG48L2hlYWQ-XG48Ym9keT5cbiAgPHNpbXBsZS1ib2R5Pjwvc2ltcGxlLWJvZHk-XG48L2JvZHk-XG4ifSx7Im5hbWUiOiJwYWNrYWdlLmpzb24iLCJjb250ZW50Ijoie1xuICBcImRlcGVuZGVuY2llc1wiOiB7XG4gICAgXCJsaXRcIjogXCJeMi4wLjBcIixcbiAgICBcIkBsaXQvcmVhY3RpdmUtZWxlbWVudFwiOiBcIl4xLjAuMFwiLFxuICAgIFwibGl0LWVsZW1lbnRcIjogXCJeMy4wLjBcIixcbiAgICBcImxpdC1odG1sXCI6IFwiXjIuMC4wXCJcbiAgfVxufSIsImhpZGRlbiI6dHJ1ZX0seyJuYW1lIjoiY29ycC1zdHlsZXMuanMiLCJjb250ZW50IjoiaW1wb3J0IHtjc3N9IGZyb20gJ2xpdCc7XG5leHBvcnQgY29uc3QgY29ycFN0eWxlcyA9IGNzc2BcbiAgICAuY29ycC1idXR0b24ge1xuICAgICAgICAgICAgYmFja2dyb3VuZC1jb2xvcjogI2U3ZTdlNztcbiAgICAgICAgICAgIGNvbG9yOiBibGFjaztcbiAgICAgICAgICAgIGJvcmRlcjogbm9uZTtcbiAgICAgICAgICAgIGJvcmRlci1yYWRpdXM6IDRweDtcbiAgICAgICAgICAgIHBhZGRpbmc6IDE1cHggMzJweDtcbiAgICAgICAgICAgIGZvbnQtc2l6ZTogMTZweDtcbiAgICAgICAgICAgIG1hcmdpbjogNHB4IDJweDtcbiAgICAgICAgICAgIGN1cnNvcjogcG9pbnRlcjtcbiAgICAgICAgICB9XG4gICAgICAgICAgLmNvcnAtYnV0dG9uOmhvdmVyIHtcbiAgICAgICAgICAgIGJhY2tncm91bmQtY29sb3I6ICNhYWFhYWE7XG4gICAgICAgICAgfVxuICAgIGA7In1d)

Also, please take a look at Benny Powers' https://github.com/bennypowers/lit-css
